eng/common/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -0,0 +1,256 @@
+parameters:
+  - name: SpecRepoUrl
+    type: string
+  - name: SdkRepoUrl
+    type: string
+  - name: SpecRepoCommit
+    type: string
+    default: '$(Build.SourceBranch)'
+  - name: SdkRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: ConfigType
+    type: string
+    values:
+      - 'TypeSpec'
+      - 'OpenAPI'
+    default: 'TypeSpec'
+    displayName: 'API specification type'
+  - name: ConfigPath
+    type: string
+    default: 'specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml'
+    displayName: 'Path to API specification file (TypeSpec/OpenAPI)'
+  - name: SkipPullRequestCreation
+    type: boolean
+    default: false
+
+extends:
+  template: /eng/pipelines/templates/stages/1es-redirect.yml
+  parameters:
+    stages:
+    - stage: Build
+      displayName: 'SDK Generation'
+      jobs:
+      - job:
+        displayName: 'SDK Generation'
+
+        variables:
+          - template: /eng/pipelines/templates/variables/image.yml
+          - name: NodeVersion
+            value: '22.13.x'
+          - name: PythonVersion
+            value: '3.13'
+          - name: SpecGenSdkVersion
+            value: 'latest'
+          - name: SdkArtifactName
+            value: SDK_Artifact
+
+        pool:
+          name: $(LINUXPOOL)
+          vmImage: $(LINUXVMIMAGE)
+          os: linux
+
+        templateContext:
+          outputParentDirectory: $(System.DefaultWorkingDirectory)/out
+          outputs:
+            - output: pipelineArtifact
+              displayName: Publish SDK artifacts to Pipeline Artifacts
+              condition: and(ne(variables['ValidationResult'], ''), eq(variables['HasSDKArtifact'], 'true'))
+              artifactName: $(sdkArtifactName)
+              targetPath: "$(System.DefaultWorkingDirectory)/out/generatedSdkArtifacts"
+            - output: pipelineArtifact
+              displayName: Publish API View artifacts to Pipeline Artifacts
+              condition: and(ne(variables['ValidationResult'], ''), eq(variables['HasApiViewArtifact'], 'true'))
+              artifactName: $(ArtifactName)
+              targetPath: "$(System.DefaultWorkingDirectory)/out/sdkApiViewArtifacts"
+            - output: pipelineArtifact
+              displayName: Publish logs to Pipeline Artifacts
+              condition: ne(variables['ValidationResult'], '')
+              artifactName: "spec-gen-sdk-logs"
+              targetPath: "$(System.DefaultWorkingDirectory)/out/logs"
```

Comment:
This is the other part change: publishing all artifacts and logs, adding conditions.